core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -153,27 +156,171 @@ public static DataSource createUserStoreDataSource(RealmConfiguration realmConfi
                     JDBCRealmConstants.MIN_EVIC_TABLE_IDLE_TIME_MILLIS)));
         }
 
-        if (realmConfig.getUserStoreProperty(JDBCRealmConstants.VALIDATION_QUERY) != null) {
-            poolProperties.setValidationQuery(realmConfig.getUserStoreProperty(
-                    JDBCRealmConstants.VALIDATION_QUERY));
-            poolProperties.setTestOnBorrow(true);
+        if (StringUtils.isNotEmpty(realmConfig.getUserStoreProperty(JDBCRealmConstants.VALIDATION_QUERY)) &&
+                !realmConfig.getUserStoreProperty(JDBCRealmConstants.VALIDATION_QUERY).trim().isEmpty()) {
```

Comment:
Can we use isNotBlank()?

Reply:
The current commons-lang version in this project does not support that.

Reply:
Ok noted. Thanks for the clarification.